ui/src/components/app-form/app-form.tsx

```diff
@@ -433,321 +624,360 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
   }, [errors, scrollToFirstError]);
 
   return (
-    <form
-      id="app-form"
-      onSubmit={handleSubmit(onFormSubmit, scrollToFirstError)}
-      className="form"
-      noValidate
-    >
-      <StyledFormSection ref={appInfoRef}>
-        <Typography component="h2" variant="subtitle1">
-          App Info
-        </Typography>
-        <Controller
-          name="display_name"
-          control={control}
-          rules={{ required: true }}
-          render={({ field: { ref, ...field } }) => (
-            <FormControl error={!!errors.display_name}>
-              {errors.display_name && (
-                <Box
-                  display="flex"
-                  alignItems="center"
-                  color="error.main"
-                  mb={2}
+    <>
+      <form
+        id="app-form"
+        onSubmit={(e) => {
+          e.preventDefault();
+
+          // Proceed to call handleSubmit
+          handleSubmit(onFormSubmit, scrollToFirstError)(e);
+        }}
+        className="form"
+        noValidate
+      >
+        <StyledFormSection ref={appInfoRef}>
+          <Typography component="h2" variant="subtitle1">
+            {deployOption === 'git' ? 'App Info (Git Repository)' : 'App Info'}
+          </Typography>
+
+          {deployOption === 'git' ? (
+            <>
+              {/* Git Repository URL Input */}
+              <Controller
+                name="repository.url"
+                control={control}
+                render={({ field }) => (
+                  <FormControl error={!!error && shouldValidate}>
+                    <TextField
+                      {...field}
+                      inputRef={repoUrlRef}
+                      id="repository.url"
+                      label="Git Repository URL"
+                      placeholder="Enter Git repository URL"
```

Comment:
I think a placeholder like the following would be useful:

```
https://github.com/nebari-dev/jhub-apps-from-git-repo-example.git
```

Reply:
Good idea. Added the sample placeholder.